### PR TITLE
Add usage events for certificate issuance and additional session types

### PR DESCRIPTION
**Pull request by timothyb89**
This adds a new usage event for certificate issuance, emitted whenever a new user certificate is generated. These events include an anonymized username, TTL, and basic info about which requests are present on the certificate (e.g. db/app/k8s requests).

Additionally, this captures additional session event types from the audit log, including database, app, and desktop sessions, and also properly tags k8s exec sessions which were previously grouped together with SSH.

### CODE REVIEW

**Review by michellescripts — APPROVED** [review]

File: lib/events/usageevents/usageevents.go [hunk 1]
```diff
@@ -67,10 +67,30 @@ func (u *UsageLogger) reportAuditEvent(ctx context.Context, event apievents.Audi
 			ConnectorType: e.Method,
 		}))
 	case *apievents.SessionStart:
-		// Note: session.start is only SSH.
+		// Note: session.start is only SSH and Kubernetes.
+		sessionType := types.SSHSessionKind
```
Comment (espadolini) [inline lib/events/usageevents/usageevents.go:71]: This should be `ssh_v2` or something along those lines, as we can't really trust `ssh` to mean SSH sessions now. 😢

Either that, or we should have a `UsageSessionStartV2` event (same underlying type, just a different oneof entry).

File: lib/services/usagereporter.go [hunk 2]
```diff
@@ -263,6 +263,25 @@ func (u *UsageUIRecoveryCodesPrintClick) Anonymize(a utils.Anonymizer) prehogv1.
 	}
 }
 
+// UsageCertificateIssuedEvent is an event emitted when a certificate has been
+// issued, used to track the duration and restriction.
+type UsageCertificateIssuedEvent prehogv1.CertificateIssuedEvent
```
Comment (espadolini) [inline lib/services/usagereporter.go:268]: Should these newtypes have the `Event` suffix or not? It's about an even split right now, we should pick one.

**Review by espadolini — COMMENTED** [review]
LGTM but we should have a way to distinguish bot/human certs (as per my comment in gravitational/cloud#3221) and we should also figure out how to convey "yes, this is actually a ssh session, not an unspecified ssh/kube session" in `UsageSessionStart`

File: lib/events/usageevents/usageevents.go (hunk 1 above)
Comment (timothyb89) [reply lib/events/usageevents/usageevents.go:71]: That's reasonable, I've added a new enum variant for V2 and switched everything over to it.

File: lib/services/usagereporter.go (hunk 2 above)
Comment (timothyb89) [reply lib/services/usagereporter.go]: I've removed it at least from this new event.

File: lib/prehog/proto/prehog/v1alpha/teleport.proto [hunk 3]
```diff
@@ -107,6 +126,8 @@ message SubmitEventRequest {
     UserLoginEvent user_login = 3;
     SSOCreateEvent sso_create = 4;
     ResourceCreateEvent resource_create = 5;
+
+    // REMOVE IN V14: Use session_start_v2 instead
```
Comment (espadolini) [inline lib/prehog/proto/prehog/v1alpha/teleport.proto:130]: We can remove this way sooner than v14, we can remove it as soon as the last tenant on cloud uses the new event.

**Review by espadolini — APPROVED** [review]

File: lib/prehog/proto/prehog/v1alpha/teleport.proto (hunk 3 above)
Comment (timothyb89) [reply lib/prehog/proto/prehog/v1alpha/teleport.proto:201]: I figure it's more of a "remove by v14 at the latest" so we can remove it earlier if we'd like... in any case it's part of the prehog proto so it's hard to change from this end :(

**Review by jakule — APPROVED** [review]

**Pull request merged**